backend/impl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitMacheteRepository.java

```diff
@@ -374,15 +374,15 @@ IGitMacheteRepositorySnapshot createSnapshot(BranchLayout branchLayout) throws G
       }
     }
 
-    private Map<String, IManagedBranchSnapshot> createManagedBranchByNameMap(List<RootManagedBranchSnapshot> rootBranches) {
-      Map<String, IManagedBranchSnapshot> branchByName = HashMap.empty();
-      Queue<IManagedBranchSnapshot> queue = Queue.ofAll(rootBranches);
-      // BFS over all branches
-      while (queue.nonEmpty()) {
-        val headAndTail = queue.dequeue();
-        val branch = headAndTail._1;
+    private LinkedHashMap<String, IManagedBranchSnapshot> createManagedBranchByNameMap(
+        List<RootManagedBranchSnapshot> rootBranches) {
+      LinkedHashMap<String, IManagedBranchSnapshot> branchByName = LinkedHashMap.empty();
+      List<IManagedBranchSnapshot> stack = List.ofAll(rootBranches);
+      // Non-recursive DFS over all branches
```

Comment:
nit:
1. **A** non-recursive DFS over all branches
2. maybe add a comment about why we use `LinkedHashMap` (saying that it is ordered collection and will guarantee us the order of branches in the end)